core/base/v7/inc/ROOT/TDirectory.h

```diff
@@ -96,47 +92,51 @@ class TDirectory {
   /// \param name  - Key of the object.
   /// \param args  - arguments to be passed to the constructor of `T`
   template <class T, class... ARGS>
-  std::weak_ptr<T> Create(const std::string& name, ARGS... args) {
-    auto ptr = std::make_shared<ToContentType_t<T>>(std::forward<ARGS...>(args...));
+  decltype(auto) Create(const std::string& name, ARGS&&... args) {
```

Comment:
Commits: I have a preference for adding new commits: it lets me see how you addressed the comments. I might then still (partially) squash or re-order the commits in the end.

`weak_ptr`: esp for the main interfaces I believe you are right, we should just give users a `shared_ptr`.